core/mo-services-impl/nmf-platform-generic-impl/src/main/java/esa/mo/platform/impl/provider/gen/GPSProviderServiceImpl.java

```diff
@@ -522,89 +557,133 @@ public GetLastKnownPositionAndVelocityResponse getLastKnownPositionAndVelocity(
-          timeOfCurrentPositionAndVelocity = System.currentTimeMillis();
-        }
+      if(useTLEpropagation)
+      {
+        Calendar targetDate = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        SpacecraftState state = getSpacecraftState(targetDate);
+        Vector3D pos = state.getPVCoordinates().getPosition();
+        position = new VectorD3D(pos.getX(), pos.getY(), pos.getZ());
+
+        positionDeviation = new VectorF3D(0f, 0f, 0f);
+
+        Vector3D velocity3D = state.getPVCoordinates().getVelocity();
+        velocity = new VectorD3D(velocity3D.getX(), velocity3D.getY(), velocity3D.getZ());
 
-        interaction.sendResponse(position, positionDeviation, velocity, velocityDeviation);
+        velocityDeviation = new VectorF3D(0f, 0f, 0f);
+      }
+      else {
+        String bestxyz = adapter.getBestXYZSentence();
+
+        String[] fields = HelperGPS.getDataFieldsFromBestXYZ(bestxyz);
+
+        position = new VectorD3D(
+                Double.parseDouble(fields[HelperGPS.BESTXYZ_FIELD.PX]),
+                Double.parseDouble(fields[HelperGPS.BESTXYZ_FIELD.PY]),
+                Double.parseDouble(fields[HelperGPS.BESTXYZ_FIELD.PZ])
+        );
+
+        positionDeviation = new VectorF3D(
+                Float.parseFloat(fields[HelperGPS.BESTXYZ_FIELD.PX_DEVIATION]),
+                Float.parseFloat(fields[HelperGPS.BESTXYZ_FIELD.PY_DEVIATION]),
+                Float.parseFloat(fields[HelperGPS.BESTXYZ_FIELD.PZ_DEVIATION])
+        );
+
+        velocity = new VectorD3D(
+                Double.parseDouble(fields[HelperGPS.BESTXYZ_FIELD.VX]),
+                Double.parseDouble(fields[HelperGPS.BESTXYZ_FIELD.VY]),
+                Double.parseDouble(fields[HelperGPS.BESTXYZ_FIELD.VZ])
+        );
+
+        velocityDeviation = new VectorF3D(
+                Float.parseFloat(fields[HelperGPS.BESTXYZ_FIELD.VX_DEVIATION]),
+                Float.parseFloat(fields[HelperGPS.BESTXYZ_FIELD.VY_DEVIATION]),
+                Float.parseFloat(fields[HelperGPS.BESTXYZ_FIELD.VZ_DEVIATION])
+        );
+      }
 
-      } catch (IOException | NumberFormatException e) {
-        interaction
-                .sendError(new MALStandardError(PlatformHelper.DEVICE_NOT_AVAILABLE_ERROR_NUMBER, null));
-        e.printStackTrace();
+      synchronized (MUTEX) { // Store the latest Position
+        currentCartesianPosition = position;
+        currentCartesianPositionDeviation = positionDeviation;
+        currentCartesianVelocity = velocity;
+        currentCartesianVelocityDeviation = velocityDeviation;
+        timeOfCurrentPositionAndVelocity = System.currentTimeMillis();
       }
+      positionAndVelocityVectors = new PositionAndVelocityVectors(
+          position, positionDeviation, velocity, velocityDeviation);
+      positionAndVelocityVectors.setThrowException(false);
 
+    } catch (IOException | NumberFormatException e) {
+      positionAndVelocityVectors.setThrowException(true);
```

Comment:
Just rethrow the `e` instead of using boolean to indicate an error. If needed, you could wrap it into another exception format just so the callers expect it. Or simply declare this method as `throws IOException, NumberFormatException`